utils/util_test.go

```diff
@@ -27,3 +28,35 @@ func TestBitsBytesConvert(t *testing.T) {
 		}
 	}
 }
+
+func TestIntToBytes(t *testing.T) {
+	numInt := 42
+	b := IntToBytes(numInt)
+	if int(binary.LittleEndian.Uint32(b)) != numInt {
+		t.Fatal("Conversion to bytes looks wrong!")
+	}
+	// TODO It is possible to call `IntToBytes` with a signed int < 0
+	// isn't that problematic?
+	// for instance this will fail:
+	// b := IntToBytes(-42)
+	// if int(binary.LittleEndian.Uint32(b)) != numInt {
```

Comment:
I think it should be `if int32(binary.LittleEndian.Uint32(b)) != numInt`. If so, the test will pass. I forgot that `int` int 64-bit system is 8 bytes.